core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -82,19 +82,38 @@ class GroupMetadataManager(brokerId: Int,
 
   this.logIdent = "[Group Metadata Manager on Broker " + brokerId + "]: "
 
-  newGauge("NumOffsets",
+  private def recreateGauge[T](name: String, gauge: Gauge[T]): Gauge[T] = {
```

Comment:
The issue is basically:

1. GroupMetadataManager1 is created.  It calls newGauge("NumOffsets"), which registers a metric with the global registry and returns it.
2. GroupMetadataManager2 is created.  It calls newGauge("NumOffsets").  However, this function looks at the global registry and finds that there is already a metric there with that name.  So it returns the gauge created in step #1.
3. The step #1 gauge is bound to the GroupMetadataManager object from step #1, not the one from step #2.  So the tests fail because their modifications to GroupMetadataManager have no effect on the metrics.

The best way to solve this would probably be to explicitly pass the metric registry to the constructor of GroupMetadataManager, rather than relying on magical global variables.  We could perhaps have it default to the global registry if no value was passed.